build.gradle

```diff
@@ -168,8 +171,10 @@ subprojects { subproject ->
 
 
 	task updateCopyrights {
-		onlyIf { !System.getenv('TRAVIS') && !System.getenv('bamboo_buildKey') }
-		inputs.files(modifiedFiles.filter { f -> f.path.contains(subproject.name) })
+		onlyIf { grgit != null && !System.getenv('TRAVIS') && !System.getenv('bamboo_buildKey') }
+		if (grgit != null) {
```

Comment:
Somehow we need this check independently of the `onlyIf` block.
Apparently Gradle builds all the options for tasks even if it is going to be disabled.

Reply:
Still the same problem
```
$ ./gradlew clean check -x test
fatal: not a git repository (or any of the parent directories): .git

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.0/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 15s
21 actionable tasks: 19 executed, 2 up-to-date
Build scan background action failed.
org.gradle.process.internal.ExecException: Process 'command 'git'' finished with non-zero exit value 128
        at org.gradle.process.internal.DefaultExecHandle$ExecResultImpl.assertNormalExitValue(DefaultExecHandle.java:417)
```

Reply:
Maybe this has to be optional somehow
```
id 'org.ajoberstar.grgit' version '3.1.1'
```

Perhaps detect the presence of `.git` ?

Reply:
This is my output:
```
C:\tmp\spring-kafka>gradlew clean check -x test

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.0/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 15s
21 actionable tasks: 18 executed, 3 up-to-date
C:\tmp\spring-kafka>    
```

As you see I have copied project to the `/tmp` and I definitely removed `.git` and `.github` dirs.
Anything else I have missed?

Reply:
I downloaded the zip from GitHub (master) then copied `build.gradle` from your PR and got the result above.

Reply:
Still works for me 🤷‍♂ , but I'll push a check against `.git` dir.

Reply:
Still fails for me; here's the full stack trace in case that helps...
```
$ ./gradlew clean check -x test
Starting a Gradle Daemon, 1 incompatible Daemon could not be reused, use --status for details
fatal: not a git repository (or any of the parent directories): .git

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.0/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 28s
21 actionable tasks: 21 executed
Build scan background action failed.
org.gradle.process.internal.ExecException: Process 'command 'git'' finished with non-zero exit value 128
        at org.gradle.process.internal.DefaultExecHandle$ExecResultImpl.assertNormalExitValue(DefaultExecHandle.java:417)
        at org.gradle.process.internal.DefaultExecAction.execute(DefaultExecAction.java:38)
        at org.gradle.process.internal.DefaultExecActionFactory.exec(DefaultExecActionFactory.java:152)
        at org.gradle.groovy.scripts.DefaultScript.exec(DefaultScript.java:249)
        at org.gradle.api.Script$exec.callCurrent(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:156)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:168)
        at build_scan_user_data_282dv3jzi832pxvzogo9mix53.execAndGetStdout(/Users/grussell/Downloads/spring-kafka-master/gradle/build-scan-user-data.gradle:73)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:101)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
        at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:483)
        at org.gradle.internal.metaobject.BeanDynamicObject.tryInvokeMethod(BeanDynamicObject.java:195)
        at org.gradle.groovy.scripts.BasicScript$ScriptDynamicObject.tryInvokeMethod(BasicScript.java:130)
        at org.gradle.internal.metaobject.ConfigureDelegate.invokeMethod(ConfigureDelegate.java:77)
        at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeOnDelegationObjects(ClosureMetaClass.java:397)
        at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:337)
        at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:64)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:156)
        at build_scan_user_data_282dv3jzi832pxvzogo9mix53$_addGitMetadata_closure1.doCall(/Users/grussell/Downloads/spring-kafka-master/gradle/build-scan-user-data.gradle:26)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:101)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
        at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:263)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1041)
        at groovy.lang.Closure.call(Closure.java:405)
        at groovy.lang.Closure.call(Closure.java:421)
        at org.gradle.util.ClosureBackedAction.execute(ClosureBackedAction.java:71)
        at org.gradle.util.ConfigureUtil.configureTarget(ConfigureUtil.java:154)
        at org.gradle.util.ConfigureUtil.configure(ConfigureUtil.java:105)
        at org.gradle.util.ConfigureUtil$WrappedConfigureAction.execute(ConfigureUtil.java:166)
        at com.gradle.scan.plugin.internal.api.j.a(SourceFile:22)
        at com.gradle.scan.plugin.internal.api.j$$Lambda$213.0000000000000000.execute(Unknown Source)
        at com.gradle.scan.plugin.internal.api.k$a.a(SourceFile:112)
        at com.gradle.scan.plugin.internal.api.k$a$$Lambda$215.0000000000000000.execute(Unknown Source)
        at com.gradle.scan.plugin.internal.api.h.a(SourceFile:61)
        at com.gradle.scan.plugin.internal.api.h$$Lambda$216.0000000000000000.run(Unknown Source)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:819)
```

Reply:
>   at build_scan_user_data_282dv3jzi832pxvzogo9mix53$_addGitMetadata_closure1.doCall(/Users/grussell/Downloads/spring-kafka-master/gradle/build-scan-user-data.gradle:26)

I think you have something for that script what is not enabled on my side.
So, doesn't look like the problem we try to solve in this issue.

Does it say anything to you?

Reply:
Doh; yes of course; environment variable:

```
GRADLE_ENTERPRISE_URL=https://ge.spring.io/scans
```

```
$ unset GRADLE_ENTERPRISE_URL 

$ ./gradlew clean check -x test
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.0/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 30s
21 actionable tasks: 21 executed

```

Merging - thanks!

Reply:
Doh; yes of course; environment variable:

```
GRADLE_ENTERPRISE_URL=https://...
```

```
$ unset GRADLE_ENTERPRISE_URL 

$ ./gradlew clean check -x test
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.0/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 30s
21 actionable tasks: 21 executed

```

Merging - thanks!